File: Templates/cmake/Fortran/OpenMP/hello_omp.f90
!==============================================================
! Copyright © Intel Corporation
!
! SPDX-License-Identifier: MIT
!=============================================================

PROGRAM test_omp_fortran
    use omp_lib
    implicit none

    print *, "Hello, OpenMP C World!"
    !$omp parallel
        print *, "  I am thread", omp_get_thread_num()
    !$omp end parallel

    print *, "All done, bye."
END PROGRAM test_omp_fortran

File: Templates/cmake/C/OpenMP/hello_omp.c
//==============================================================
// Copyright © Intel Corporation
//
// SPDX-License-Identifier: MIT
// =============================================================

#include <omp.h>
#include <stdio.h>

int main(int argc, char *argv[]) {
  printf("Hello, OpenMP C World!\n");
#pragma omp parallel
  { printf("  I am thread %d\n", omp_get_thread_num()); }
  printf("All done, bye.\n");
  return 0;
}
